app/src/test/java/org/simple/clinic/sync/indicator/SyncIndicatorViewControllerTest.kt

```diff
@@ -123,26 +126,37 @@ class SyncIndicatorViewControllerTest {
     verify(dataSync).sync(SyncGroup.FREQUENT)
   }
 
+  data class ShowSyncErrorParams(
+      val error: ResolvedError,
+      val shouldShowError: Boolean
+  )
+
   @Test
   @Parameters(method = "params for testing sync errors")
-  fun `when sync indicator is clicked and sync starts, appropriate failure dialog should show if any sync error is thrown`(error: ResolvedError) {
+  fun `appropriate failure dialog should be shown for sync errors`(testCase: ShowSyncErrorParams) {
+    val (error: ResolvedError,
+        shouldShowError: Boolean) = testCase
+
     whenever(dataSync.streamSyncErrors()).thenReturn(Observable.just(error))
     whenever(dataSync.sync(SyncGroup.FREQUENT)).thenReturn(Completable.complete())
 
     lastSyncStateStream.onNext(LastSyncedState(lastSyncProgress = SUCCESS))
     uiEvents.onNext(SyncIndicatorViewClicked)
 
-    verify(dataSync).sync(SyncGroup.FREQUENT)
-    verify(dataSync).streamSyncErrors()
-    verify(indicator).showErrorDialog(errorType = error)
+    if (shouldShowError) {
```

Comment:
Tests should not have conditionals, these should at least be split into two parameterized tests. One for the `if` part and the other for the `else` part.

Reply:
I'll see if I can improve this test.